components/channel_members_rhs/channel_members_rhs.tsx

```diff
@@ -88,10 +106,32 @@ export default function ChannelMembersRHS({channel, searchTerms, membersCount, c
 
         actions.setChannelMembersRhsSearchTerm('');
         setEditing(false);
-        actions.loadProfilesAndReloadChannelMembers(channel.id);
+        setPage(0);
+        setLoadingMore(false);
+
+        actions.loadProfilesAndReloadChannelMembers(0, USERS_PER_PAGE, channel.id);
         actions.loadMyChannelMemberAndRole(channel.id);
     }, [channel.id, channel.type]);
 
+    const searchDebounced = debounce(
+        async () => {
+            if (searchTerms) {
+                setEditing(false);
+                setPage(0);
+                setLoadingMore(false);
+
+                await actions.searchProfilesAndChannelMembers(searchTerms, {in_team_id: channel.team_id, in_channel_id: channel.id});
+                actions.loadProfilesAndReloadChannelMembers(0, USERS_PER_PAGE, channel.id);
+                actions.loadMyChannelMemberAndRole(channel.id);
```

Comment:
Do these need to be reloaded when we're searching? I would think we only need the search results at this point